tests/sentry/sentry_metrics/test_gen_metrics_multiprocess_steps.py

```diff
@@ -0,0 +1,618 @@
+
+    """
+    formatted_payload = (
+        json.dumps(invalid_payload).encode("utf-8") if format_payload else invalid_payload
+    )
+    message_batch = [
+        Message(
+            BrokerValue(
+                KafkaPayload(None, json.dumps(counter_payloads[0]).encode("utf-8"), []),
+                Partition(Topic("topic"), 0),
+                0,
+                BROKER_TIMESTAMP,
+            )
+        ),
+        Message(
+            BrokerValue(
+                KafkaPayload(None, formatted_payload, []),
+                Partition(Topic("topic"), 0),
+                1,
+                BROKER_TIMESTAMP,
+            )
+        ),
+    ]
+    # the outer message uses the last message's partition, offset, and timestamp
+    last = message_batch[-1]
+
+    outer_message = Message(Value(message_batch, last.committable))
+
+    with caplog.at_level(logging.ERROR):
+        new_batch = MESSAGE_PROCESSOR.process_messages(outer_message=outer_message)
+
+    # we expect just the valid counter_payload msg to be left
+    expected_msg = message_batch[0]
+    expected_new_batch = [
+        Message(
+            Value(
+                KafkaPayload(
+                    None,
+                    json.dumps(__translated_payload(counter_payloads[0])).encode("utf-8"),
+                    [("metric_type", b"c")],
+                ),
+                expected_msg.committable,
+            )
+        )
+    ]
+    compare_message_batches_ignoring_metadata(new_batch, expected_new_batch)
+    assert error_text in caplog.text
+
+
+@pytest.mark.django_db
+def test_process_messages_rate_limited(caplog, settings) -> None:
+    """
+    Test handling of `None`-values coming from the indexer service, which
+    happens when postgres writes are being rate-limited.
+    """
+    settings.SENTRY_METRICS_INDEXER_DEBUG_LOG_SAMPLE_RATE = 1.0
+    rate_limited_payload = deepcopy(distribution_payloads[0])
+    rate_limited_payload["tags"]["rate_limited_test"] = "true"
+
+    message_batch = [
+        Message(
+            BrokerValue(
+                KafkaPayload(None, json.dumps(counter_payloads[0]).encode("utf-8"), []),
+                Partition(Topic("topic"), 0),
+                0,
+                BROKER_TIMESTAMP,
+            )
+        ),
+        Message(
+            BrokerValue(
+                KafkaPayload(None, json.dumps(rate_limited_payload).encode("utf-8"), []),
+                Partition(Topic("topic"), 0),
+                1,
+                BROKER_TIMESTAMP,
+            )
+        ),
+    ]
+    # the outer message uses the last message's partition, offset, and timestamp
+    last = message_batch[-1]
+    outer_message = Message(Value(message_batch, last.committable))
+
+    message_processor = MessageProcessor(
+        get_ingest_config(UseCaseKey.PERFORMANCE, IndexerStorage.MOCK)
+    )
+    # Insert a None-value into the mock-indexer to simulate a rate-limit.
+    mock_indexer: Any = message_processor._indexer
```

Comment:
this seems odd -- what was the error here before?  I don't think annotating this as `Any` should fix anything here

Reply:
It was
```
tests/sentry/sentry_metrics/test_gen_metrics_multiprocess_steps.py:531: error: "StringIndexer" has no attribute "indexer"  [attr-defined]
```
I am guessing it's due to some downcasting